src/operations/indexes.ts

```diff
@@ -458,39 +418,25 @@ export class IndexExistsOperation extends AbstractCallbackOperation<boolean> {
     this.indexes = indexes;
   }
 
-  override executeCallback(
-    server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<boolean>
-  ): void {
+  override async execute(server: Server, session: ClientSession | undefined): Promise<boolean> {
     const coll = this.collection;
     const indexes = this.indexes;
 
-    indexInformation(
-      coll.s.db,
-      coll.collectionName,
-      { ...this.options, readPreference: this.readPreference, session },
-      (err, indexInformation) => {
-        // If we have an error return
-        if (err != null) return callback(err);
-        // Let's check for the index names
-        if (!Array.isArray(indexes)) return callback(undefined, indexInformation[indexes] != null);
-        // Check in list of indexes
-        for (let i = 0; i < indexes.length; i++) {
-          if (indexInformation[indexes[i]] == null) {
-            return callback(undefined, false);
-          }
-        }
-
-        // All keys found return true
-        return callback(undefined, true);
-      }
-    );
+    const info = await indexInformation(coll.s.db, coll.collectionName, {
+      ...this.options,
+      readPreference: this.readPreference,
+      session
+    });
+    // Let's check for the index names
+    if (!Array.isArray(indexes)) return info[indexes] != null;
+
+    // All keys found return true
+    return Array.prototype.every(indexName => info[indexName] != null);
```

Comment:
```suggestion
    return indexes.every(indexName => info[indexName] != null);
```

I can't find documentation about what this returns but in my testing it always returns true.  I don't see tests failing because of this - can you add some basic testing for the `index exists` operation that would have caught this bug?  I think 

- a test for an index that doesn't exist returns false
- a test for an index that does exist returns true
- a test for an array of indexes that exist returns true
- a test for an array of indexes where one does't exist returns false

is probably sufficient